src/Coordination/KeeperStorage.cpp

```diff
@@ -3302,8 +3239,15 @@ KeeperStorage<Container>::ResponsesForSessions KeeperStorage<Container>::process
 
     {
         std::lock_guard lock(transaction_mutex);
+
         if (new_last_zxid)
+        {
+            if (!keeper_context->digestEnabledOnCommit())
+                nodes_digest = uncommitted_transactions.front().nodes_digest.value;
```

Comment:
I got an impression that it should be like this instead, is it a mistake?
```suggestion
            if (keeper_context->digestEnabledOnCommit())
                nodes_digest = uncommitted_transactions.front().nodes_digest.value;
```

Reply:
It's a bit confusing, but if calculation of digest on commit is disabled, than we take the digest we calculated during preprocessing (i.e. we take from uncommitted transaction)